datafusion/functions-nested/src/array_has.rs

```diff
@@ -215,7 +215,11 @@ fn array_has_dispatch_for_array<O: OffsetSizeTrait>(
         let needle_row = Scalar::new(needle.slice(i, 1));
         let eq_array = compare_with_eq(&arr, &needle_row, is_nested)?;
         let is_contained = eq_array.true_count() > 0;
-        boolean_builder.append_value(is_contained)
+        if is_contained || eq_array.null_count() == 0 {
```

Comment:
I am surprised this isn't 

```suggestion
        if is_contained && eq_array.null_count() == 0 {
```

I thought if the eq_array is null that means there was at least one comparison that is not known 🤔

Reply:
The semantics are basically:
- if the element is contained, return true
- otherwise if there is a null value in the array, return null
- otherwise return false

So nulls don't matter if there's match, it only matters if there's no match